packages/ts/react-crud/test/autogrid.spec.tsx

```diff
@@ -745,6 +734,98 @@ describe('@hilla/react-crud', () => {
+        const StreetRenderer = ({ item }: { item: Person }): JSX.Element => (
+          <span>{item.address?.street.toUpperCase()}</span>
+        );
+        const grid = await GridController.init(
+          render(
+            <TestAutoGrid
+              columnOptions={{
+                firstName: { renderer: NameRenderer },
+                'address.street': { renderer: StreetRenderer },
+              }}
+            />,
+          ),
+          user,
+        );
+        await assertColumns(
+          grid,
+          'firstName',
+          'lastName',
+          'gender',
+          'email',
+          'someInteger',
+          'someDecimal',
+          'vip',
+          'birthDate',
+          'shiftStart',
+          'appointmentTime',
+          'address.street',
+          'address.city',
+          'address.country',
+          'department',
+        );
+        expect(grid.getBodyCellContent(0, 0)).to.have.rendered.text('JANE');
+        expect(grid.getBodyCellContent(0, 1)).to.have.rendered.text('Love');
+        expect(grid.getBodyCellContent(0, 10)).to.have.rendered.text('122 SOUTH STREET');
+        expect(grid.getBodyCellContent(0, 11)).to.have.rendered.text('South Town');
+        // The header filter was not overridden
+        const cell = grid.getHeaderCellContent(1, 0);
+        expect(cell.firstElementChild!.querySelector('vaadin-text-field')).to.exist;
+      });
+
+      it('respects the header setting from custom column options', async () => {
+        // With header filters
+        let result = render(<TestAutoGrid columnOptions={{ firstName: { header: 'FIRSTNAME' } }} />);
+        let grid = await GridController.init(result, user);
+        expect(grid.getHeaderCellContent(0, 0).innerText).to.equal('FIRSTNAME');
+
+        // Without header filters
+        result.unmount();
+        result = render(<TestAutoGrid noHeaderFilters columnOptions={{ firstName: { header: 'FIRSTNAME' } }} />);
+        grid = await GridController.init(result, user);
+        expect(grid.getHeaderCellContent(0, 0).innerText).to.equal('FIRSTNAME');
+      });
+
+      it('renders row numbers if requested', async () => {
+        const grid = await GridController.init(render(<TestAutoGrid rowNumbers />), user);
+        await assertColumns(
+          grid,
+          '',
+          'firstName',
+          'lastName',
+          'gender',
+          'email',
+          'someInteger',
+          'someDecimal',
+          'vip',
+          'birthDate',
+          'shiftStart',
+          'appointmentTime',
+          'address.street',
+          'address.city',
+          'address.country',
+          'department',
+        );
+        expect(grid.getBodyCellContent(0, 0)).to.have.rendered.text('1');
+        expect(grid.getBodyCell(0, 0).style.flexGrow).to.equal('0');
+      });
+    });
+
+    describe('custom columns', () => {
```

Comment:
I only moved the custom column tests into a separate test suite, and added two new tests at the end.